### PR TITLE
Task/day card list

### PR DESCRIPTION
**Pull request by Pavel-Yarakhovich**
Usefull notes being made while working on this PR:

1.  you can't destructure props as it harms reactivity
2. changing object prop didn't cause child component's rerender. Instead, I had to use the `createComputed()` inside that child component
3. `createSelector()` is nice

### CODE REVIEW

File: src/components/day-card-list/day-card.tsx [hunk 1]
```diff
@@ -1,7 +1,86 @@
...
 };
 
+interface DayCardProps {
+  day: DayType;
+  onCardClick: (dayId: number) => void;
+  isClicked: boolean;
+}
+
+const DayCard: Component<DayCardProps> = props => {
+  const [preparedData, setPreparedData] = createSignal<CardData>();
```
Comment (pavel-kru) [inline src/components/day-card-list/day-card.tsx]: do we actually need `const [preparedData, setPreparedData] = createSignal<CardData>();`?

File: src/components/day-card-list/day-card.tsx [hunk 2]
```diff
@@ -1,7 +1,86 @@
...
+    data.date = format(props.day.dt * 1000, "iii',' PPP',' p");
+
+    Object.entries(props.day.main).forEach(([key, value]) => {
+      const formattedKey = key.split('_').join(' ');
+      data.weather.push({ [formattedKey]: value });
+    });
+
+    setPreparedData(data);
+  });
+
```
Comment (pavel-kru) [inline src/components/day-card-list/day-card.tsx:44]: we could just return data:

`  const cardData = createMemo(() => {
    const data: CardData = {
      date: '',
      weather: [],
    };

    data.date = format(props.day.dt * 1000, "iii',' PPP',' p");

    Object.entries(props.day.main).forEach(([key, value]) => {
      const formattedKey = key.split('_').join(' ');
      data.weather.push({ [formattedKey]: value });
    });

    return data;
  });`

File: src/components/day-card-list/day-card.tsx [hunk 3]
```diff
@@ -1,7 +1,86 @@
...
+  return (
+    <div
+      class={
+        props.isClicked
+          ? `${cardStylesBasic} ${cardStylesHovered}`
+          : cardStylesBasic
+      }
+      onClick={() => props.onCardClick(props.day.id)}
+    >
+      <Show when={preparedData()} keyed>
```
Comment (pavel-kru) [inline src/components/day-card-list/day-card.tsx]: then we could use cardData() insted of   preparedData()

File: src/components/day-card-list/day-card-list.tsx [hunk 4]
```diff
@@ -1,7 +1,64 @@
...
+  const isClicked = createSelector(selectedCardId);
+
+  return (
+    <>
+      <div class="flex gap-4 mb-2">
+        <For each={props.list}>
+          {(day, i) => (
+            <DayCard
+              day={day}
+              onCardClick={dayId => {
```
Comment (pavel-kru) [inline src/components/day-card-list/day-card-list.tsx]: wow. 

not sure, I understood the full idea.
could we just set card on click?

Comment (pavel-kru) [reply src/components/day-card-list/day-card-list.tsx]: could we do
`
const DayCardList: Component<DayCardListProps> = props => {
  const [selectedCard, setSelectedCard] = createSignal<DayType | null>(null, {
    equals: false,
  });

  const isClicked = createSelector(selectedCard);

  return (
    <>
      <div class="flex gap-4 mb-2">
        <For each={props.list}>
          {(day, i) => (
            <DayCard
              day={day}
              `
             onCardClick={card => {
                if (!selectedCard()) {
                  setSelectedCard(card);
                } else {
                  setSelectedCard(null);
                }
              }}
              isClicked={isClicked(day)}
            />
`
          )}
        </For>
      </div>

      <Show when={selectedCard()} keyed>
        {card => <DetailedDayCard day={card} />}
      </Show>
    </>
  );
};
`


// src/components/day-card-list/day-card.tsx

interface DayCardProps {
  day: DayType;
  `onCardClick: (day: DayType) => void;`
  isClicked: boolean;
}


.....
  return (
    <div
      class={
        props.isClicked
          ? `${cardStylesBasic} ${cardStylesHovered}`
          : cardStylesBasic
      }
      `onClick={() => props.onCardClick(props.day)}`
    >
.....

File: src/views/Today.tsx [hunk 5]
```diff
@@ -1,7 +1,26 @@
...
+
+// Components
+import { DayCardList } from '../components';
+
+interface TodayProps {
+  forecast: DayType | unknown;
+}
+
+const Today: Component<TodayProps> = props => {
+  const [list, setList] = createSignal<any>([]);
```
Comment (pavel-kru) [inline src/views/Today.tsx]: interesting feature createComputed.

maybe it is better to replace lines 13 - 17 with
`  const list = createMemo(() => (props.forecast ? [props.forecast] : []));
`

File: tailwind.config.js [hunk 6]
```diff
@@ -5,7 +5,11 @@ module.exports = {
...
     extend: {},
-    colors,
+    colors: {
+      ...colors,
+      'main-bg': '#49e6ff1c',
+      'day-card-bg': '#ffffff',
+    },
   },
   plugins: [],
 };
```
Comment (pavel-kru) [inline tailwind.config.js:15]: In common, great work! Interesting features.

**Review by pavel-kru — CHANGES_REQUESTED** [review]

**Comment by pavel-kru** [comment]
> 



> Usefull notes being made while working on this PR:
> 
> 1. you can't destructure props as it harms reactivity
> 2. changing object prop didn't cause child component's rerender. Instead, I had to use the `createComputed()` inside that child component
> 3. `createSelector()` is nice

Cool, interesting! nice research

File: src/components/day-card-list/day-card-list.tsx (hunk 4 above)
Comment (Pavel-Yarakhovich) [reply src/components/day-card-list/day-card-list.tsx]: Yes, absolutely. The idea was just to use createEffect to see how it works ))

File: src/components/day-card-list/day-card.tsx (hunk 2 above)
Comment (Pavel-Yarakhovich) [reply src/components/day-card-list/day-card.tsx:44]: Yes, this was my first version. But it didn't work for me for some reason. So, I used signal as a workaround.

File: src/views/Today.tsx (hunk 5 above)
Comment (Pavel-Yarakhovich) [reply src/views/Today.tsx]: Why not. It should work as well. I believe. Again, tried to use as much of API as possible.

**Pull request merged**